hugr-core/src/std_extensions/collections/static_array.rs

```diff
@@ -0,0 +1,396 @@
+//! An extension for working with static arrays.
+use std::{
+    hash, iter,
+    sync::{self, Arc},
+};
+
+use crate::{
+    builder::{BuildError, Dataflow},
+    extension::{
+        prelude::{option_type, usize_t},
+        resolution::{ExtensionResolutionError, WeakExtensionRegistry},
+        simple_op::{
+            try_from_name, HasConcrete, HasDef, MakeExtensionOp, MakeOpDef, MakeRegisteredOp,
+            OpLoadError,
+        },
+        ExtensionId, ExtensionSet, OpDef, SignatureError, SignatureFunc, TypeDef,
+    },
+    ops::{
+        constant::{CustomConst, TryHash, ValueName},
+        ExtensionOp, NamedOp, OpName, Value,
+    },
+    types::{
+        type_param::{TypeArgError, TypeParam},
+        CustomType, PolyFuncType, Signature, Type, TypeArg, TypeBound, TypeName,
+    },
+    Extension, Wire,
+};
+
+use super::array::ArrayValue;
+
+use delegate::delegate;
+use lazy_static::lazy_static;
+
+/// Reported unique name of the extension
+pub const EXTENSION_ID: ExtensionId = ExtensionId::new_static_unchecked("collections.static_array");
+/// Reported unique name of the array type.
+pub const STATIC_ARRAY_TYPENAME: TypeName = TypeName::new_inline("static_array");
+/// Extension version.
+pub const VERSION: semver::Version = semver::Version::new(0, 1, 0);
+
+#[derive(Debug, Clone, serde::Serialize, serde::Deserialize, PartialEq, derive_more::From)]
+/// Statically sized array of values, all of the same type.
+pub struct StaticArrayValue {
```

Comment:
Need to check that serde serialisation is space efficient enough for O(100.000) elements